src/xref2/compile.ml

```diff
@@ -101,9 +101,14 @@ and class_type_expr env =
   | Signature s -> Signature (class_signature env s)
 
 and class_type env c =
+  let exception Compile_class_type in
```

Comment:
What's the purpose of using locally defined exceptions?

Reply:
They are meant to be easy to grep and remove and can only be caught by catch-all handlers, increasing the odds of making them fatal.